olp-cpp-sdk-core/src/http/winhttp/NetworkWinHttp.cpp

```diff
@@ -0,0 +1,1006 @@
+          WinHttpConnect(http_session_, host_name.data(), port, 0);
+      if (!connection->http_connection) {
+        return SendOutcome(ErrorCode::OFFLINE_ERROR);
+      }
+    }
+
+    connection->last_used = GetTickCount64();
+
+    handle =
+        std::make_shared<RequestData>(id, connection, callback, header_callback,
+                                      data_callback, payload, request);
+    http_requests_[id] = handle;
+  }
+
+  DWORD flags = (url_components.nScheme == INTERNET_SCHEME_HTTPS)
+                    ? WINHTTP_FLAG_SECURE
+                    : 0;
+  LPWSTR http_verb = L"GET";
+  if (request.GetVerb() == NetworkRequest::HttpVerb::POST) {
+    http_verb = L"POST";
+  } else if (request.GetVerb() == NetworkRequest::HttpVerb::PUT) {
+    http_verb = L"PUT";
+  } else if (request.GetVerb() == NetworkRequest::HttpVerb::HEAD) {
+    http_verb = L"HEAD";
+  } else if (request.GetVerb() == NetworkRequest::HttpVerb::DEL) {
+    http_verb = L"DELETE";
+  } else if (request.GetVerb() == NetworkRequest::HttpVerb::PATCH) {
+    http_verb = L"PATCH";
+  }
+
+  LPCSTR content = WINHTTP_NO_REQUEST_DATA;
+  DWORD content_length = 0;
+
+  if (request.GetVerb() != NetworkRequest::HttpVerb::HEAD &&
+      request.GetVerb() != NetworkRequest::HttpVerb::GET &&
+      request.GetBody() != nullptr && !request.GetBody()->empty()) {
+    content = (LPCSTR) & (request.GetBody()->front());
+    content_length = (DWORD)request.GetBody()->size();
+  }
+
+  /* Create a request */
+  auto http_request =
+      WinHttpOpenRequest(handle->connection_data->http_connection, http_verb,
+                         url_components.lpszUrlPath, NULL, WINHTTP_NO_REFERER,
+                         WINHTTP_DEFAULT_ACCEPT_TYPES, flags);
+  if (!http_request) {
+    EDGE_SDK_LOG_ERROR(LOGTAG, "WinHttpOpenRequest failed " << GetLastError());
+
+    std::unique_lock<std::recursive_mutex> lock(mutex_);
+
+    http_requests_.erase(id);
+    return SendOutcome(ErrorCode::IO_ERROR);
+  }
+
+  const auto& network_settings = request.GetSettings();
+
+  WinHttpSetTimeouts(http_request,
+                     network_settings.GetConnectionTimeout() * 1000,
+                     network_settings.GetConnectionTimeout() * 1000,
+                     network_settings.GetTransferTimeout() * 1000,
+                     network_settings.GetTransferTimeout() * 1000);
+
+  const auto& proxy = network_settings.GetProxySettings();
+
+  if (proxy.GetType() != NetworkProxySettings::Type::NONE) {
+    std::wstring proxy_string = ProxyString(proxy);
+
+    WINHTTP_PROXY_INFO proxy_info;
+    proxy_info.dwAccessType = WINHTTP_ACCESS_TYPE_NAMED_PROXY;
+    proxy_info.lpszProxy = const_cast<LPWSTR>(proxy_string.c_str());
+    proxy_info.lpszProxyBypass = WINHTTP_NO_PROXY_BYPASS;
+
+    DWORD error_code = ERROR_SUCCESS;
+    if (!WinHttpSetOption(http_request, WINHTTP_OPTION_PROXY, &proxy_info,
+                          sizeof(proxy_info))) {
+      error_code = GetLastError();
```

Comment:
```suggestion
```
Better to create this variable in the LOG function itself, cause the `error_code` is not used outside the LOG methods